axop/plugins.py

```diff
@@ -45,6 +74,9 @@ def get_payload(self, context: str, plugin: dict) -> dict:
             if i == "context":
                 values["context"] = context
                 continue
+            if i == "model" and self.is_encoded(plugin[i]):
```

Comment:
The main issue here is that we hard code "model" keywork as a possible b64/zlib value.
model is not a reserved keyword like "context" in Axon API and it is possible that a plugin can set a model with an integer type (just an exemple).

It should be better to enhanced the variables definition.
https://github.com/croixbleueqc/axonserver-operator/blob/e79d75715dd795d0b11144af5910b15882671dee/axop/typing/plugins.py#L29

This is a List of string but maybe that should be a List of string OR List of "new object".
This "new object" can define the variable name, encoding (eg: base64:zlib), something else ?

By doing so, it will not be necessary:
- to hardcode "model" or any other plugin variables
- to check encoding before decoding => we can just do it as expected and trigger a ValueError if there is a decoding issue

Reply:
I updated the code to do as you specified